chainntnfs/txnotifier.go

```diff
@@ -272,8 +285,17 @@ type SpendRequest struct {
 // NewSpendRequest creates a request for a spend notification of either an
 // outpoint or output script. A nil outpoint or an allocated ZeroOutPoint can be
 // used to dispatch the confirmation notification on the script.
-func NewSpendRequest(op *wire.OutPoint, pkScript []byte) (SpendRequest, error) {
+func NewSpendRequest(op *wire.OutPoint, pkScript []byte,
+	heightHint uint32) (SpendRequest, error) {
+
 	var r SpendRequest
+	if len(pkScript) == 0 {
+		return r, errors.New("an output script must be provided")
+	}
+	if heightHint == 0 {
+		return r, errors.New("a height hint greater than 0 must be " +
+			"provided")
+	}
```

Comment:
Something looks as if it is not right here. This function takes two parameters that is does nothing with other than checking. Shouldn't those parameters be part of the `SpendRequest` struct too? Or is there a different design that doesn't show this friction? Same for function above.

Reply:
I just decided to group the validation checks within these functions to prevent having another function as these are already doing some checks as well. We don't want to store these in the structs because they're used as keys for maps and we wouldn't be able to coalesce different requests for the same txid/outpoint/script. I'm not opposed to creating a separate function for them though if you think it's a cleaner approach.

Reply:
It looked into this part of the code a bit more. It seems that the code in the three `RegisterConfirmationsNtfn` functions up to the `txNotifier.RegisterConf` call is duplicated. What would you think about moving that part into `txNotifier` and doing the parameter validation there as well?

Reply:
Done.